src/strands/experimental/bidirectional_streaming/agent/agent.py

```diff
@@ -8,32 +8,31 @@
 Key capabilities:
 - Persistent conversation sessions with concurrent processing
 - Real-time audio input/output streaming
-- Mid-conversation interruption and tool execution
+- Automatic interruption detection and tool execution
 - Event-driven communication with model providers
 """
 
 import asyncio
 import json
 import logging
-import random
-from concurrent.futures import ThreadPoolExecutor
-from typing import Any, AsyncIterable, Callable, Mapping, Optional
+from typing import Any, AsyncIterable, Mapping, Optional, Union
```

Comment:
nit: Lets use built in typing where possible
```suggestion
from typing import Any, AsyncIterable, Mapping
```